### ISSUE TITLE
External references to schemas do not count towards schema being in use

### ISSUE THREAD

**Issue by tsjondin** [Redocly/openapi-cli#97]
Not entirely sure if this is a bug or a quirk in the specification but, when referencing a schema from an external file that has been "sourced" by another ref, that schema is still classified as being unused, i.e. no-unused-schemas

The schema has been and can be successfully resolved since removing that schema definition causes a resolve-ref error

It feels as if no-unused-schemas is only checking for local uses.

(running swagger-cli validate on the same structure causes no errors, but I prefer the output of openapi over swagger-cli)

Example where the reference to index.yaml#/components/schemas/Test from TestPath.yaml can be resolved but the schema is still not classified as "in-use" by the validator :

index.yaml:
```yaml
openapi: 3.0.0
info:
  title: 'API'
  version: 1.0.0
  contact:
    name: 'Contact Name'
    email: 'contact@mail.com'
  license:
    name: 'MIT License'
    url: 'https://opensource.org/licenses/MIT'
servers:
  -
    url: 'http://localhost:3000'
paths:
  '/test':
    $ref: './TestPath.yaml'
components:
  schemas:
    Test:
      $ref: './Test.yaml'
```

Test.yaml:
```yaml
properties:
  id:
    type: string
    format: uuid
```

TestPath.yaml:
```yaml
get:
  summary: 'Test'
  description: 'Test'
  operationId: getTest
  responses:
    '200':
      description: 'Successfully fetched test'
      content:
        application/json:
          schema:
            $ref: './index.yaml#/components/schemas/Test'
```

**Comment by RomanHotsiy** [comment]
It is a bug. We'll look into this.

Note that you don't need to reference you schema from `components.schemas`.

If you use `bundle` command it will automatically pull the schema to the components..

**Comment by tsjondin** [comment]
Great! And thanks for the quick reply :)

Yes I could reference the component schema directly from the TestPath.yaml (just do a `$ref: './Test.yaml'`), but this was just a minimal example, in my larger api specification the directories are structured differently and I want a central manifest of all components which will not use the schemas, hence cause this warning (though I could disable the check for now in the config).

Or were you thinking of something else?

I don't really want to bundle my specification while working with it, though I can run something along the line of: `openapi bundle index.yaml > /tmp/bundled.yaml && openapi validate /tmp/bundled.yaml`as a workaround

**Comment by adamaltman** [comment]
I tried this in 1.0.0-beta.1 and received this error:

```
validating index.yaml...
[1] index.yaml:20:7 at #/components/schemas/Test/$ref

Property `$ref` is not expected here.

Did you mean: oneOf ?

18 | schemas:
19 |   Test:
20 |     $ref: './Test.yaml'
   |     ^^^^

Error was generated by the spec rule.
```

This seems like a bug to me. @RomanHotsiy ?


Field Name | Type | Description
-- | -- | --
schemas | Map[string, Schema Object \| Reference Object] | An object to hold reusable Schema Objects.

**Comment by RomanHotsiy** [comment]
@adamaltman this was resolver issue. I've fixed it. I will close it once release a new beta later today.

**Comment by RomanHotsiy** [comment]
Should be fixed with 1.0.0-beta.2
